app/models/translation.rb

```diff
@@ -39,6 +40,22 @@ def delete_draft!
     return :bad_request
   end
 
+  def translated_pages
+    translated_pages = []
+
+    resource.pages.each do |resource_page|
+      custom_pages.each do |custom_page|
+        if resource_page.id == custom_page.page_id
+          translated_pages.push(custom_page)
+        else
+          translated_pages.push(resource_page)
+        end
+      end
+    end
+
+    translated_pages
```

Comment:
I could re-write this:
```
resource.pages.map do |resource_page|
  custom_pages.map do |custom_page|
    next custom_page if resource_page.id == custom_page.page_id
    resource_page
  end
end.flatten
```
But I think that you end up with an array that is filled with a lot of duplicates. What are you trying to accomplish here?